### PR TITLE
Convert SearchResult, InteractiveAuth, PushProcessor and Scheduler to TS

### PR DESCRIPTION
**Pull request by t3chguy**
Requires https://github.com/matrix-org/matrix-js-sdk/pull/1772

Preview: https://611438cffaac191aea354d58--matrix-react-sdk.netlify.app
⚠️ Do you trust the author of this PR? Maybe this build will steal your keys or give you malware. Exercise caution. Use test accounts.

### CODE REVIEW

**Comment by t3chguy** [comment]
Blocked until after RC

File: src/components/views/dialogs/DeactivateAccountDialog.tsx [hunk 1]
```diff
@@ -180,7 +184,7 @@ export default class DeactivateAccountDialog extends React.Component<IProps, ISt
                     <InteractiveAuth
                         matrixClient={MatrixClientPeg.get()}
                         authData={this.state.authData}
-                        makeRequest={this.onUIAuthComplete}
+                        makeRequest={this.onUIAuthComplete as any}
```
Comment (turt2live) [inline src/components/views/dialogs/DeactivateAccountDialog.tsx:187]: Comment or something to denote why we're using `any`, with possible means to correct our bad behaviour.

tldr: `// TODO: Types`

**Review by turt2live — APPROVED** [review]
This side lgtm
